pandas/_testing/__init__.py

```diff
@@ -292,21 +296,40 @@ def makeBoolIndex(k=10, name=None):
     return Index([False, True] + [False] * (k - 2), name=name)
 
 
+def makeNumericIndex(k=10, name=None, *, dtype):
+    dtype = pandas_dtype(dtype)
+
+    if dtype.kind == "i":
+        values = list(range(k))
+    elif dtype.kind == "u":
+        start_num = 2 ** (dtype.itemsize * 8 - 1)
+        values = [start_num + i for i in range(k)]
+    elif dtype.kind == "f":
+        values = sorted(np.random.random_sample(k)) - np.random.random_sample(1)
+        values = values * (10 ** np.random.randint(0, 9))
+    else:
+        raise NotImplementedError(f"wrong dtype {dtype}")
+
+    return Index(values, dtype=dtype, name=name)
```

Comment:
In #41153 we'll only need to change this to `return NumericIndex(values, dtype=dtype, name=name)`.